app/controllers/index.js

```diff
@@ -1,7 +1,74 @@
 import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
 
 export default class IndexController extends Controller {
   sort = '-created';
   page = 0;
   size = 20;
+
+  @service store;
+  @service router;
+
+  @tracked showModal = false;
+  @tracked name = '';
+  @tracked description = ``;
+  @tracked duplicateNames = [];
+
+  modelName = 'generated-form';
+  hasBeenFocused = false;
+
+  @action
+  closeModal() {
+    this.name = '';
+    this.description = ``;
+    this.showModal = false;
+    this.hasBeenFocused = false;
+  }
+
+  @action
+  async handleNameChange(event) {
+    this.name = event.target.value;
```

Comment:
I would trim the value before checking for duplicates. Otherwise they might sidestep the validation by adding spaces.